src/lua/datetime.lua

```diff
@@ -0,0 +1,611 @@
+        error('only one of nsec, usec or msecs may defined simultaneously', 2)
+    end
+    if usec ~= nil then
+        nsec = usec * 1e3
+    elseif msec ~= nil then
+        nsec = msec * 1e6
+    end
+    local ts = obj.timestamp
+    if ts ~= nil then
+        local fraction
+        s, fraction = math_modf(ts)
+        -- if there are separate nsec, usec, or msec provided then
+        -- timestamp should be integer
+        if count_usec == 0 then
+            nsec = fraction * 1e9
+        elseif fraction ~= 0 then
+            error('only integer values allowed in timestamp '..
+                  'if nsec, usec, or msecs provided', 2)
+        end
+        hms = true
+    end
+
+    local offset = obj.tzoffset
+    if offset ~= nil then
+        offset = get_timezone(offset)
+        check_range(offset, -720, 720, offset)
+    end
+
+    if obj.tz ~= nil then
+        nyi('tz')
+    end
+
+    -- .year, .month, .day
+    if ymd then
+        y = y or 1970
+        M = M or 1
+        d = d or 1
+        if d < 0 then
+            d = builtin.tnt_dt_days_in_month(y, M)
+        end
+        dt = builtin.tnt_dt_from_ymd(y, M, d)
+    end
+
+    -- .hour, .minute, .second
+    local secs = 0
+    if hms then
+        secs = (h or 0) * 3600 + (m or 0) * 60 + (s or 0)
+    end
+
+    return datetime_new_dt(dt, secs, nsec, offset or 0)
+end
+
+--[[
+    Convert to text datetime values
+
+    - datetime will use ISO-8601 format:
+        1970-01-01T00:00Z
+        2021-08-18T16:57:08.981725+03:00
+]]
+local function datetime_tostring(self)
+    check_date(self, 'datetime.tostring()')
+
+    local buff = ffi.new('char[?]', TOSTRING_BUFSIZE)
+    local len = builtin.datetime_to_string(buff, TOSTRING_BUFSIZE, self)
+    assert(len < TOSTRING_BUFSIZE)
+    return ffi.string(buff)
+end
+
+--[[
+    Dispatch function to create datetime from string or table.
+    Creates default timeobject (pointing to Epoch date) if
+    called without arguments.
+]]
+local function datetime_from(o)
+    if o == nil or type(o) == 'table' then
+        return datetime_new(o)
+    end
+end
+
+--[[
+    Create datetime object representing current time using microseconds
+    platform timer and local timezone information.
+]]
+local function local_now()
+    local d = datetime_new_raw(0, 0, 0)
+    builtin.datetime_now(d)
+    return d
+end
+
+--[[
+    Return table in os.date('*t') format, but with timezone
+    and nanoseconds
+]]
+local function datetime_totable(self)
+    local secs = local_secs(self) -- hour:minute should be in local timezone
+    local dt = local_dt(self)
+
+    return {
+        year = builtin.tnt_dt_year(dt),
+        month = builtin.tnt_dt_month(dt),
+        yday = builtin.tnt_dt_doy(dt),
+        day = builtin.tnt_dt_dom(dt),
+        wday = (ffi.cast('int32_t', builtin.tnt_dt_dow(dt)) + 1) % 7,
+        hour = math_floor((secs / 3600) % 24),
+        min = math_floor((secs / 60) % 60),
+        sec = secs % 60,
+        isdst = false,
```

Comment:
Is DST (daylight saving time) active. This information will only available since the moment we integrate Olson database, in some of other commits. [G]LIBC does have some weak heuristics basing on global `tzname[]` content, but it's platform/os specific and quite unreliable (e.g. there is different value I see in Debian and Fedora). 
At the moment, when there is no human-readable timezone facilities, anything beyond false in DST is making no much sense. It will be defined properly, after Olson and proper timezone parsing.